conda_build/build.py

```diff
@@ -288,6 +290,21 @@ def write_about_json(m, config):
             value = m.get_value('about/%s' % key)
             if value:
                 d[key] = value
+        # for sake of reproducibility, record some conda info
+        conda_info = json.loads(subprocess.check_output(['conda', 'info', '--json', '-s']))
+        d['conda_version'] = conda_info['conda_version']
+        d['conda_build_version'] = conda_info['conda_build_version']
+        d['conda_env_version'] = conda_info['conda_env_version']
+        d['offline'] = conda_info['offline']
+        d['channels'] = conda_info['channels']
```

Comment:
This will contain channels with the private token will it not?

Reply:
It will contain http credentials of the channels, see #3991, e.g. 
```
{
  "channels": [
    "https://USER:XXXXXXXX@example.com/conda",
    "conda-forge",
    "defaults"
  ],
```